### PR TITLE
[*] MO : So Colissimo version 2.9.20

### PR DESCRIPTION
**Pull request by quadrateam**
Release for socolissimo 2.9.20 
 - Bug fix for calling API La poste in SSL for prestashop  1.5 and >
- Add 3 new fields in table deliveries_info : lotacheminement, distributionsort, versionplantri

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Your new version is 2.9.20. So your upgrade should be linked to this one, not 2.9.19. Can you please update it ?

**Comment by quadrateam** [comment]
sorry but we had to package a temporary release for some hurry customers, (v2.9.19 in the changelog) and we prefer to avoid conflict between delivered versions. 
Also we already have a v2.9.21 which is already actually in beta testing.
Thanks for your understanding

File: views/templates/front/socolissimo_redirect.tpl [hunk 1]
```diff
@@ -32,50 +32,53 @@
...
-	}
+		function serialiseInput(inputs) {
+			var str = '?first_call=1&';
+			for (var cle in inputs) {
+				str += cle + '=' + inputs[cle] + '&';
+			}
+			if (!soBwdCompat)
+				return (str + 'gift=' + $('#gift').attr('checked') + '&gift_message=' + $('#gift_message').attr('value'));
+			else
+				return (str + 'module=socolissimo&controller=redirect&fc=module&gift=' + $('#gift').attr('checked') + '&gift_message=' + $('#gift_message').attr('value'));
```
Comment (Quetzacoalt91) [inline views/templates/front/socolissimo_redirect.tpl]: Did you test this URL when the URL rewriting is enabled ?

File: views/templates/front/socolissimo_redirect_mobile.tpl [hunk 2]
```diff
@@ -31,39 +31,42 @@
...
+
+		function serialiseInput(inputs) {
+			var str = '?first_call=1&';
+			for (var cle in inputs) {
+				str += cle + '=' + inputs[cle] + '&';
+			}
+			if (!soBwdCompat)
+				return (str + 'gift=' + $('#gift').attr('checked') + '&gift_message=' + $('#gift_message').attr('value'));
+			else
+				return (str + 'module=socolissimo&controller=redirectmobile&fc=module&gift=' + $('#gift').attr('checked') + '&gift_message=' + $('#gift_message').attr('value'));
```
Comment (Quetzacoalt91) [inline views/templates/front/socolissimo_redirect_mobile.tpl]: Did you also test this URL when the URL rewriting is enabled ?

**Comment by Quetzacoalt91** [comment]
Hi @quadrateam,

You should check anyway your upgrade. When I try to execute it, I get the following response:
capture du 2015-06-02 14 52 48

If you confirm me that it works anyway ? Thanks

**Comment by quadrateam** [comment]
Hi @Quetzacoalt91 

The url rewriting support is now included (this part was originally scheduled in the release which is a non-sense).

We will investigate for the install report.

**Comment by quadrateam** [comment]
@Quetzacoalt91,

We have modified the runUpgrade function to let prestashop apply the upgrade if prestashop's version is superior to 1.5. (problem was duplicate upgrade for prestashop 1.5 + )

Now upgrade will no more bug.

Regards,

**Comment by Quetzacoalt91** [comment]
Hi @quadrateam,

I have just tested with the lastest commits, and now the upgrade works properly. Thanks a lot.

The upgrade has just been merged and the changes should be available from tomorrow.

Best regards

**Pull request merged**